modules/renter/contractor/persist.go

```diff
@@ -120,6 +133,26 @@ func (c *Contractor) load() error {
 		}
 	}
 
+	// COMPATv1.1.0
+	//
+	// Wwhen loading old contracts, the corresponding relationships may not be
+	// known. Use the hostdb to make sure the relationship set is fully filled
+	// out.
+	allHosts := c.hdb.AllHosts()
+	hmap := make(map[modules.NetAddress]modules.HostDBEntry)
+	// Iterate backwards (lowest score to highest) so that in the event of
+	// duplicate hosts for a netaddress, the highest score host is the one in
+	// the map.
+	for i := len(allHosts) - 1; i >= 0; i-- {
+		hmap[allHosts[i].NetAddress] = allHosts[i]
+	}
```

Comment:
No because this is really only a hack around the renter's dependency on the NetAddress type. Really we should not be filtering by NetAddress at all, everything should be pubkey based.

I would write a sanity check and also a test but the weight function is not exported.